packages/drivers/odsp-driver/src/compactSnapshotParser.ts

```diff
@@ -116,50 +116,92 @@ function readTreeSection(node: NodeCore) {
 		 */
 		const length = treeNode.length;
 		if (length > 0 && treeNode.getMaybeString(0) === "name") {
-			if (length === 4) {
-				const content = treeNode.getMaybeString(2);
-				// "name": <node name>
-				// "children": <blob id>
-				if (content === "children") {
-					const result = readTreeSection(treeNode.getNode(3));
-					trees[treeNode.getString(1)] = result.snapshotTree;
-					slowTreeStructureCount += result.slowTreeStructureCount;
-					continue;
+			switch (length) {
```

Comment:
>witch (length) { [](http://example.com/codeflow?start=4&length=16)

@jatgarg Can this whole switch block be replaced by something like

Reply:
The reason is to avoid a bit of duplication and consolidating the fetching of groupId, Children, unreferenced and value fields. This may not apply as is, may need a couple of tweaks but sharing as a general idea. 

PS: line 143-144 may be tricky to squeeze in, but you are more familiar with this code, so hope you can figure that out faster.

Reply:
So, we expect the fields to be at certain place as per the contract and wants to read those fields from there only. What switch is saying that we will read them anywhere they are which is not correct. For example, unreferenced if present would be at 3rd index only as specified in structure above each of those if.

Reply:
Thanks for sharing that. Do you expect the fields to be received in a different order and if they are in different order, you'd want to skip reading those? For example, if `unreferenced` in the input is at 4th index, you'd want to completely skip reading it? I am trying to understand if the order matters and if it does, can the input come in a different order and what do we do when it is out of order?

PS: This is just for my knowledge. I saw the code is for parsing a snapshot tree and it's not trying to validate the order, but just find properties inside the treeNode at specific indexes. So curious about why does the order matter while parsing?

Reply:
So, one thing is about position which we expect to be pre-determined index according to contract but we still read them if they are out of order. But the other thing is grouping where we expect unreferenced only in case tree/alongside "children" prop and not for blobs, "nodeType" would only be there if "value" is there. So, although if sections, lead to a bit of duplicity but that is more concreate for the reader and comments also mention the structure that we are trying to read.

Reply:
Ah I see! Cool, I think I have more to learn about this parsing. Thank you for answering Jatin.